### PR TITLE
Fix for Issue #19: Does not support new Origen app dir structure.

### LINKED ISSUE #19: Does not support new Origen app dir structure
Apps that use Origen `app/` dir workspace structure appears to have issues with this plug-in.  Pages generated for have incorrect links in them.  

As an example I am trying to render `doc_helpers/searchable.html` in order to implement a searchable set of guides. I have to pass in a valid `:root` option to the render otherwise I'll get a compile error:

    % opts = options.merge(index: index, root: "#{Origen.root}/app/templates/web/guides/", pdf_title: "My Guides")
    % render "doc_helpers/searchable.html", options.merge(opts) do
    <%= yield %>

    % end
    
However then my links include `app/templates/web/guides` which is incorrect.  

Older apps pass in this: 

    % opts = options.merge(index: index, root: "#{Origen.root}/templates/web/guides/", pdf_title: "My Guides")
    % render "doc_helpers/searchable.html", options.merge(opts) do
    <%= yield %>

    % end
    
and yet the doc helpers is able to correctly render the internal links to `latest/` and not `/templates/web/guides/`.   

I suspect it's a matter of having the plugin know which type of app (old or new structure) it is dealing with.

### PR DESCRIPTION
**Pull request by redxeth**
Does not support new Origen app dir structure #19

Added check for if app contains `app/` dir to get proper `templates/` relative reference.

### CODE REVIEW

File: lib/origen_doc_helpers/pdf.rb [hunk 1]
```diff
@@ -71,6 +75,7 @@ def generate_pdf
         end
       end
       cmd += "#{pdf_output_dir}/#{filename}"
+puts "cmd=#{cmd}"
```
Comment (pderouen) [inline lib/origen_doc_helpers/pdf.rb]: Probably didn't intend to keep this line

**Comment by pderouen** [comment]
Wow, there were more changes needed than I expected. It would have been nice if there was 1 central "get the templates directory" method.

I'm good with the updates once the lint error is fixed and the debug line removed.

**Comment by redxeth** [comment]
Yeah-- there is some related code built in to Origen (https://github.com/Origen-SDK/origen/blob/master/lib/origen/file_handler.rb#L210) where it appears to use `clean_path_to` but you still have to pass it both options.  Anyway, this solution works fine I think.

File: lib/origen_doc_helpers/pdf.rb (hunk 1 above)
Comment (redxeth) [reply lib/origen_doc_helpers/pdf.rb]: Yeah -- fixed thanks

**Review by pderouen — APPROVED** [review]

**Pull request merged**